src/lucky/errors.cr

```diff
@@ -7,8 +7,10 @@ module Lucky
 
   # Raised when a route could not be found
   class RouteNotFoundError < Error
-    def initialize(method : String, path : String)
-      super "Could not find route matching #{method} #{path}"
+    property context : HTTP::Server::Context
+
+    def initialize(@context : HTTP::Server::Context)
```

Comment:
I think this could also accept the params so it is easier to get at specific params when needed: `@params : Lucky::Params` would be the type signature